### PR TITLE
Task reminder

### PR DESCRIPTION
**Pull request by ywrsusan**
*Issue number of the reported bug or feature request: #<2835>*
https://jira.prod.bloomberg.com/browse/RDISCROWD-2835

Change other task setting UI - use "Save" button to save and remove "Cancel" button

Screen Shot 2020-03-18 at 5 31 30 PM
Screen Shot 2020-03-18 at 5 30 26 PM

### CODE REVIEW

File: templates/projects/task_settings.html [hunk 1]
```diff
@@ -26,5 +26,8 @@
     <div id="answer_fields" class="col-md-4">
         {{ render_project_card_option(project, upload_method, title=_('Answer Fields'), explanation=_('Configure Answer Fields'), link=url_for('project.answerfieldsconfig', short_name=project.short_name), link_action_text=_('Answer Fields'), icon='pencil-square-o')}}
     </div>
+    <div id="progress_reminder" class="col-md-4">
+        {{ render_project_card_option(project, upload_method, title=_('Progress Reminder'), explanation=_('Configure reminder on project progress'), link=url_for('project.task_progress_reminder', short_name=project.short_name), link_action_text=_('Reminder'), icon='fa fa-bell-o ')}}
```
Comment (kbecker42) [inline templates/projects/task_settings.html]: Should we consider renaming "Progress Reminder", perhaps to "Task Notifications"?

File: templates/projects/progress_reminder.html [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{% extends "projects/base.html" %}
+{% set active_page = "projects" %}
+{% set active_link  = "tasks" %}
+{% set section = _('Progress Reminder') %}
```
Comment (kbecker42) [inline templates/projects/progress_reminder.html]: If we decide to rename the page title, we may want to update this as well to "Task Notifications" or similar wording.

The filename could be updated as well from `progress_reminder.html` to `task_notifications.html`. :)

File: templates/projects/progress_reminder.html [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+{% from "_formhelpers.html" import render_field %}
+
+{% block projectcontent %}
+<h3>Send email notification on the progress of tasks</h3>
+<form id="task_progress_reminder" method="post" action="{{ url_for('project.task_progress_reminder', short_name=project.short_name) }}">
```
Comment (kbecker42) [inline templates/projects/progress_reminder.html]: Just noting `task_progress_reminder`, in case the page is renamed. Just fyi.

File: templates/projects/progress_reminder.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{% extends "projects/base.html" %}
+{% set active_page = "projects" %}
+{% set active_link  = "tasks" %}
+{% set section = _('Progress Reminder') %}
+{% from "_formhelpers.html" import render_field %}
+
+{% block projectcontent %}
+<h3>Send email notification on the progress of tasks</h3>
```
Comment (kbecker42) [inline templates/projects/progress_reminder.html]: *Fyi, this could be held off until we take on the Webhook URL notification task.*

As we may be adding adding notification options to this form, should we change this to a checkbox and rename it to "Enable email notification"?

This way, when we add additional options (such as the webhook), we would include subsequent checkboxes for each option. For example:

```
[X] Enable email notification
[X] Enable Webhook URL.
     [__________________________]

Notify when number of incomplete tasks drop to:
[_____ 14 _____]

[Set] | [Cancel]
```

Just brainstorming UI ideas. Thanks.

Comment (dchhabda) [reply templates/projects/progress_reminder.html]: Adding checkbox would require validations to apply sync between checkboxes and associated fields; disable field if checkbox unchecked, delete field values when checkbox value changed from checked to unchecked... 
As this page is dedicated to notifications, imo, current solution w/o checkbox would work - honor values when set, else ignore.

Comment (kbecker42) [reply templates/projects/progress_reminder.html]: That makes sense. However, when we do eventually add additional notifications options (webhook) we may need a way to disable the email notification option, as there is no specific email input field to differentiate when it should be active or disabled. :)

i.e., the text box for "Notify when the number of incomplete tasks ..." would apply to both email and webhook notifications.

Comment (dchhabda) [reply templates/projects/progress_reminder.html]: I'm fine whichever way feature owners would expect this. Will add my notes to jira and we can discuss with team.

Comment (ywrsusan) [reply templates/projects/progress_reminder.html]: Or maybe rephrase the description as 
"notify when the number of incomplete tasks drops to the following number (leave empty to disable)"
same as webhook.

Comment (kbecker42) [reply templates/projects/progress_reminder.html]: Well, the idea is that once we have additional notification options, the user will want to be able to selectively toggle off/on specific notification methods. The user may only want the webhook, but not the email. However, for all methods, the textbox value would be required.

We can re-visit this redesign when we add more fields. :)

Until then, if we could just make the label more precise (i.e., "Notify when the number of remaining tasks is less than or equal to"). I'm not sure if it's necessary to mention leaving the value blank to disable (perhaps, that is implied?),
